chain/src/main/scala/org/bitcoins/chain/blockchain/sync/ChainSync.scala

```diff
@@ -0,0 +1,110 @@
+package org.bitcoins.chain.blockchain.sync
+
+import org.bitcoins.chain.api.ChainApi
+import org.bitcoins.chain.blockchain.ChainHandler
+import org.bitcoins.chain.models.BlockHeaderDb
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+import org.bitcoins.core.protocol.blockchain.BlockHeader
+import org.bitcoins.core.util.BitcoinSLogger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ChainSync extends BitcoinSLogger {
+
+  /** This method checks if our chain handler has the tip of the blockchain as an external source
+    * If we do not have the same chain, we sync our chain handler until we are at the same best block hash
+    * @param chainHandler our internal chain handler
+    * @param getBlockHeaderFunc a function that we can call to retrieve a block
+    * @param getBestBlockHashFunc a function that can call a third party source (bitcoind, block explorer etc)
+    *                             to retrieve what the best block is on the blockchain
+    * @param ec
+    * @return
+    */
+  def sync(chainHandler: ChainHandler,
+           getBlockHeaderFunc: DoubleSha256DigestBE => Future[BlockHeader],
+           getBestBlockHashFunc: () => Future[DoubleSha256DigestBE])(implicit ec: ExecutionContext): Future[ChainApi] = {
```

Comment:
Would it work to change the definition to `getBestBlockHashFunc: => Future[DoubleSha256DigestBE])`? Cleaner API, IMO